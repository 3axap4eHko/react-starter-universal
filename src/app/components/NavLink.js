import React from 'react';
import withStyles from 'react-jss';
import classNames from 'classnames';
import createLink from 'react-steersman/createLink';

const styles = theme => ({
  item: {
    padding: 10,
  },
  activeItem: {
    backgroundColor: 'rgba(0,0,0,0.2)',
  },
  link: {
    color: theme.palette.text,
  },
  activeLink: {
    color: theme.palette.accent,
  },
});

export default withStyles(styles)(createLink(({ to, classes, title, navigate, match }) => (
  <li className={classNames(classes.item, { [classes.activeItem]: match })}>
    <a
      href={to}
      className={classNames(classes.link, { [classes.activeLink]: match })}
      title={title}
      onClick={navigate}
    >
      {title}
    </a>
  </li>
)));
