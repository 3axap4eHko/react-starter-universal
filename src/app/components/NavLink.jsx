'use strict';

import React, {Component, PropTypes} from 'react';
import {Link} from 'react-router';
const {any, object, string, oneOfType} = PropTypes;

class NavLink extends Component {
    static contextTypes = {
        router: object
    };
    static propTypes = {
        to: oneOfType([string, object]),
        children: any
    };
    render() {
        const {to, children, ...props} = this.props;
        const {router} = this.context;
        const isActive = router.isActive(to, true);
        return (
            <li className={isActive && 'active'}>
                <Link to={to} {...props}>{children}</Link>
            </li>
        );
    }
}

export default NavLink;