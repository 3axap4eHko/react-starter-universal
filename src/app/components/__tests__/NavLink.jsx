import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NavLink from '../NavLink';

configure({ adapter: new Adapter() });

const testUrl = '/test-url';

test('NavLink snapshot match', () => {
  const component = renderer.create(
    <BrowserRouter>
      <NavLink to={testUrl} />
    </BrowserRouter>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Feature values', () => {
  // Render a checkbox with label in the document
  const element = shallow(
    <BrowserRouter>
      <NavLink to={testUrl} />
    </BrowserRouter>,
  );
  const liTag = element.find(NavLink);
  expect(liTag.length).toEqual(1);
});