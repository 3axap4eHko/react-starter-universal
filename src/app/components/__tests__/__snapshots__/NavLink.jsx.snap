// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink snapshot match 1`] = `
<li
  className={null}
>
  <a
    href="/test-url"
    onClick={[Function]}
  />
</li>
`;
